src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -143,22 +143,6 @@ public static bool Report(Exception exception)
         [DebuggerHidden]
         public static bool ReportWithoutCrash(Exception exception)
         {
-            // There have been cases where a new, unthrown exception has been passed to this method.
```

Comment:
this will done in WatsonReporter. NonFatalHandler points to WatsonReporter in Roslyn. it is null in compiler.